### DISCUSSION TITLE
Mapping Airtable primary keys - quick question

### DISCUSSION THREAD

**Discussion by SolMi-Sera** [nocodb/nocodb#1841] [Q&A]
Trying to migrate my Airtable schema into noco and I could use some help understanding the differences that exist between noco and Airtable  regarding, what Airtable calls the "recordID".   

Case: 
• In AT, every record has a recordID assigned in the background by AT and is a valid primary key. 
• In my Base design, I also created a primary key in every table which is some meaningful concatenated value. 

-- Throughout my backend workflow, I use both of these values; meaning that 1 field, in every table of my Base, has 2 "primary keys". 

I'm trying to figure out if/how I need to handle that situation. 

Question: 
Does noco rely solely on the primary key we designate? 
Or, does it create a "recordID" in the background with every new record, like Airtable does?

-SolMi

**Comment by o1lab** [comment]
We create a primary key in background by default (and its an autoincremented value) - we plan to change this going forward. And we should have a migration tool available soon. Some of our users has successfully migrated data from airtable to postgres via sequin and have used nocodb to connect to it.

**Reply by SolMi-Sera** [reply]
Thanks for the reply, that answers my question. 

You'll have to excuse my ignorance, but is there some reason that exporting my Base metadata from AT, parsing into CSV format, and then uploading into noco wont work? 

Instead of going from: 
AT > db > noco

I was planning on going from: 
AT > csv > noco -- which I thought would  update the connected sqlite3 db for me. 

Will that work? 

I don't have experience with any db other than learning db fundamentals in Access years ago. I'm looking to transition from AT to noco for my backend because of AT's 50k record limit. 

Thanks in advance for helping me wrap my head around this.

**Reply by o1lab** [reply]
Uploading csv is currently in alpha with noco and uploading works with existing table only. However, please stay tuned as this will be seamless going forward.

You can reach out to me at naveen at nocodb dot com or [book a time](calendly.com/nocodb-meeting) that works for you - I will help to migrate.

**Reply by SolMi-Sera** [reply]
Wow, that'd be huge, I will definitely take you up on that!! Th	anks for the quick responses and willingness to help out - im getting more and more excited about about this path; excited to see where it goes.
